code/items/IBuyableItem.cs

```diff
@@ -9,21 +11,20 @@ public struct ShopItemData
         public string Name;
         public string Description;
         public int Price;
+        public SlotType SlotType;
```

Comment:
That's the only change I don't like in this PR, as Perks are `IBuyableItems` as well, but never have a `SlotType`

Reply:
I agree, I don't like this change either.

We should discuss what alternatives exist to this approach of checking if players are able to purchase a weapon/equipment when there is no slot available.

Reply:
`SlotType` still could be a part of `ICarriableItem`, then you just need to check for it explicitly.